src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/AttendanceService.java

```diff
@@ -90,17 +90,21 @@ public void updateAttendance(final long id, final UpdateAttendanceDto newAttenda
     }
 
     private Optional<Long> handleIEPWarningScenarios(final Attendance attendance, final UpdateAttendanceDto newAttendanceDetails) {
+        final var hasIEPWarning =
+                attendance.getAbsentReason() != null &&
+                        AbsentReason.getIepTriggers().contains(attendance.getAbsentReason());
+
         final var shouldTriggerIEPWarning =
                 newAttendanceDetails.getAbsentReason() != null &&
                         AbsentReason.getIepTriggers().contains(newAttendanceDetails.getAbsentReason());
 
```

Comment:
Wonder if it would be better to exit earlier? 
`
if (hasIEPWarning && shouldTriggerIEPWarning)
   return Optional.empty();
`

Reply:
Could even rename 

> hasIEPWarning
 to 
> alreadyIssuedIEPWarning
.
Making it `if (alreadyIssuedIEPWarning && shouldTriggerIEPWarning) return Optional.empty();`

Reply:
I did wonder about the naming of the variable.  I was thinking 'hasTriggeredIEPWarning' but wasn't sure whether it was too bulky.  hasIEPWarning seemed more concise but I'll stick to the conventions already in place and update it.